File: examples/python/agg-fulfillment/handlers/status.py
"""Fulfillment status enum.

Provides typed status values instead of string literals.
"""

from enum import StrEnum


class FulfillmentStatus(StrEnum):
    """Fulfillment aggregate status values."""

    PENDING = "pending"
    PICKING = "picking"
    PACKING = "packing"
    SHIPPED = "shipped"
    DELIVERED = "delivered"

File: examples/go/agg-fulfillment/logic/status.go
package logic

// FulfillmentStatus represents the status of a fulfillment in the aggregate.
type FulfillmentStatus string

const (
	// FulfillmentStatusPending indicates the shipment is pending.
	FulfillmentStatusPending FulfillmentStatus = "pending"
	// FulfillmentStatusPicking indicates items are being picked.
	FulfillmentStatusPicking FulfillmentStatus = "picking"
	// FulfillmentStatusPacking indicates items are being packed.
	FulfillmentStatusPacking FulfillmentStatus = "packing"
	// FulfillmentStatusShipped indicates the shipment has been shipped.
	FulfillmentStatusShipped FulfillmentStatus = "shipped"
	// FulfillmentStatusDelivered indicates the shipment has been delivered.
	FulfillmentStatusDelivered FulfillmentStatus = "delivered"
)

// String returns the string representation of the status.
func (s FulfillmentStatus) String() string {
	return string(s)
}
